feature/auth/src/main/java/com/teamoffroad/feature/auth/presentation/AgreeTermsAndConditionsViewModel.kt

```diff
@@ -0,0 +1,129 @@
+package com.teamoffroad.feature.auth.presentation
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.teamoffroad.feature.auth.domain.usecase.UserMarketingAgreeUseCase
+import com.teamoffroad.feature.auth.presentation.model.AgreeTermsAndConditionsUiState
+import com.teamoffroad.feature.auth.presentation.model.DialogState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AgreeTermsAndConditionsViewModel @Inject constructor(
+    private val marketingInfoUseCase: UserMarketingAgreeUseCase,
+) : ViewModel() {
+    private val _allChecked: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val allChecked: StateFlow<Boolean> = _allChecked.asStateFlow()
+    private val _serviceUtil: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val serviceUtil: StateFlow<Boolean> = _serviceUtil.asStateFlow()
+    private val _personalInfo: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val personalInfo: StateFlow<Boolean> = _personalInfo.asStateFlow()
+    private val _location: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val location: StateFlow<Boolean> = _location.asStateFlow()
+    private val _marketing: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val marketing: StateFlow<Boolean> = _marketing.asStateFlow()
+    private val _dialogState: MutableStateFlow<DialogState> = MutableStateFlow(DialogState.EMPTY)
+    val dialogState: StateFlow<DialogState> = _dialogState.asStateFlow()
+    private val _agreeTermsAndConditionsUiState: MutableStateFlow<AgreeTermsAndConditionsUiState> =
+        MutableStateFlow(
+            AgreeTermsAndConditionsUiState.EMPTY
+        )
+    val agreeTermsAndConditionsUiState: StateFlow<AgreeTermsAndConditionsUiState> =
+        _agreeTermsAndConditionsUiState.asStateFlow()
+
+    fun allCheckedChangedListener() {
+        if (_allChecked.value) {
+            _allChecked.value = false
+            _serviceUtil.value = false
+            _personalInfo.value = false
+            _location.value = false
+            _marketing.value = false
+        } else if (!_allChecked.value) {
+            _allChecked.value = true
+            _serviceUtil.value = true
+            _personalInfo.value = true
+            _location.value = true
+            _marketing.value = true
+        }
+    }
+
+    fun serviceCheckedChangedListener() {
+        _serviceUtil.value = !_serviceUtil.value
```

Comment:
```suggestion
        _serviceUtil.value = !serviceUtil.value
```
r: 값을 읽을 때는 위와 같이 사용해야 합니다! 전체적으로 수정해주시죠!

Reply:
수정했습니다!!